### PR TITLE
GH-261 Addressed Ticket Feedback.

### PR DESCRIPTION
**Pull request by milindmore22**
related to #261 ticket feedback.

### CODE REVIEW

**Comment by tomasstark** [comment]
@milindmore22 Let's please resolve conflicting files before reviewing:

- assets/build/revenue-generator-admin.js
- assets/src/js/pages/revenue-generator-paywall.js

**Comment by milindmore22** [comment]
@tomasstark resolved conflicts please review.

File: assets/src/js/pages/revenue-generator-paywall.js [hunk 1]
```diff
@@ -2296,6 +2228,119 @@ import { RevGenModal } from '../utils/rev-gen-modal';
...
+
+					switch ( currentPurchaseType ) {
+						case 'subscription':
+							newTitle += __(
+								' Subscription',
+								'revenue-generator'
+							);
+							break;
+						case 'timepass':
+							newTitle += __( ' Pass', 'revenue-generator' );
```
Comment (tomasstark) [inline assets/src/js/pages/revenue-generator-paywall.js:2305]: Same here, let's add that leading space as a non-translatable string. `' ' + __( 'Pass', 'revenue-generator' )`.

File: inc/classes/class-admin.php [hunk 2]
```diff
@@ -1072,13 +1076,15 @@ public function update_paywall() {
...
-				$subscription['access_to']   = 'all'; // @todo make it dynamic
+				$subscription['title']        = sanitize_text_field( wp_unslash( $subscription_data['title'] ) );
+				$subscription['description']  = sanitize_text_field( wp_unslash( $subscription_data['desc'] ) );
+				$subscription['price']        = floatval( $subscription_data['price'] );
+				$subscription['revenue']      = sanitize_text_field( $subscription_data['revenue'] );
+				$subscription['duration']     = sanitize_text_field( $subscription_data['duration'] );
+				$subscription['period']       = sanitize_text_field( $subscription_data['period'] );
+				$subscription['access_to']    = 'all'; // @todo make it dynamic
+				$subscription['custom_title'] = sanitize_text_field( $subscription_data['custom_title'] );
+				$subscription['custom_desc']  = sanitize_text_field( $subscription_data['custom_desc'] );
```
Comment (tomasstark) [inline inc/classes/class-admin.php:1087]: Are these changes just running `phpcs`? Doesn't appear there are any changes apart from spaces.

File: inc/classes/post-types/class-subscription.php [hunk 3]
```diff
@@ -363,6 +369,9 @@ public function update_subscription( $subscription_data ) {
 			update_post_meta( $subscription_id, '_rg_revenue', $subscription_data['revenue'] );
 			update_post_meta( $subscription_id, '_rg_duration', $subscription_data['duration'] );
 			update_post_meta( $subscription_id, '_rg_access_to', $subscription_data['access_to'] );
+			update_post_meta( $subscription_id, '_rg_period', $subscription_data['period'] );
+			update_post_meta( $subscription_id, '_rg_custom_title', $subscription_data['custom_title'] );
+			update_post_meta( $subscription_id, '_rg_custom_desc', $subscription_data['custom_desc'] );
 		}
```
Comment (tomasstark) [inline inc/classes/post-types/class-subscription.php:376]: Same here. Was something changed or is it just alignment?

File: inc/classes/post-types/class-time-pass.php [hunk 4]
```diff
@@ -364,6 +370,8 @@ public function update_time_pass( $time_pass_data ) {
 			update_post_meta( $time_pass_id, '_rg_duration', $time_pass_data['duration'] );
 			update_post_meta( $time_pass_id, '_rg_period', $time_pass_data['period'] );
 			update_post_meta( $time_pass_id, '_rg_access_to', $time_pass_data['access_to'] );
+			update_post_meta( $time_pass_id, '_rg_custom_title', $time_pass_data['custom_title'] );
+			update_post_meta( $time_pass_id, '_rg_custom_desc', $time_pass_data['custom_desc'] );
 		}
```
Comment (tomasstark) [inline inc/classes/post-types/class-time-pass.php:376]: Any changes or just spacing?

**Review by tomasstark — COMMENTED** [review]
Let's address those comments before merging it please.

File: inc/classes/class-admin.php (hunk 2 above)
Comment (milindmore22) [reply inc/classes/class-admin.php:1087]: yup this is for phpcs issues, variables need to be aligned. + two new lines at bottom.

File: inc/classes/post-types/class-subscription.php (hunk 3 above)
Comment (milindmore22) [reply inc/classes/post-types/class-subscription.php:376]: All three lines are newly added.

File: inc/classes/post-types/class-time-pass.php (hunk 4 above)
Comment (milindmore22) [reply inc/classes/post-types/class-time-pass.php:376]: two lines have been added newly

**Comment by milindmore22** [comment]
@tomasstark Addressed PR feedback please re-review.

File: inc/classes/post-types/class-subscription.php (hunk 3 above)
Comment (tomasstark) [reply inc/classes/post-types/class-subscription.php:376]: Oh I see, I got confused by diff at first. Makes sense! 👍

File: inc/classes/post-types/class-time-pass.php (hunk 4 above)
Comment (tomasstark) [reply inc/classes/post-types/class-time-pass.php:376]: Perfect!

**Review by tomasstark — APPROVED** [review]
Looking great! LGTM! 🚀

**Pull request merged**